dev/com.ibm.ws.logging/src/com/ibm/ws/logging/internal/impl/BaseTraceService.java

```diff
@@ -1639,6 +1643,52 @@ public synchronized void flush() throws IOException {
         }
     }
 
+    public static class ThreadLocalByteArrayOutputStream extends OutputStream {
+        private final static int BYTE_ARRAY_OUTPUT_BUFFER_SIZE = 128;
+        private final ThreadLocal<ByteArrayOutputStream> threadLocal = new ThreadLocal<ByteArrayOutputStream>() {
+            @Override
+            protected ByteArrayOutputStream initialValue() {
+                return new ByteArrayOutputStream(BYTE_ARRAY_OUTPUT_BUFFER_SIZE);
+            }
+        };
+
+        @Override
+        public void write(int b) throws IOException {
+            threadLocal.get().write(b);
+
+        }
+
+        public void reset() {
+            if (threadLocal.get().size() > BYTE_ARRAY_OUTPUT_BUFFER_SIZE)
+                threadLocal.remove();
+            threadLocal.get().reset();
+        }
+
+        @Override
+        public void flush() throws IOException {
+            threadLocal.get().flush();
+        }
+
+        @Override
+        public void close() throws IOException {
+            threadLocal.get().close();
+        }
+
+        @Override
+        public String toString() {
+            return threadLocal.get().toString();
+        }
+
+        public static int getByteArrayOutputThreshold() {
+            int BYTE_ARRAY_OUTPUT_THRESHOLD_BASE_CASE = 256000;
+            if ((System.getenv("STACKTRACE_BYTE_ARRAY_THRESHOLD") != null)) {
+                return Integer.valueOf(System.getenv("STACKTRACE_BYTE_ARRAY_THRESHOLD"));
+            }
```

Comment:
should catch any parsing exception and use the default value in those cases.  should also ensure the number is >= 0, else use the default.

Reply:
Fixed in [572ed02](https://github.com/OpenLiberty/open-liberty/pull/16276/commits/572ed024f574b4abf76fe1067cfe1b4152b06de4)